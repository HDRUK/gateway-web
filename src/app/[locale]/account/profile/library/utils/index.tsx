import { IconButton } from "@mui/material";
import { createColumnHelper } from "@tanstack/react-table";
import { LibraryListItem, SelectedLibrary } from "@/interfaces/Library";
import Link from "@/components/Link";
import StyledCheckbox from "@/components/StyledCheckbox";
import TooltipIcon from "@/components/TooltipIcon";
import { CheckIcon, DeleteForeverIcon } from "@/consts/icons";
import { RouteName } from "@/consts/routeName";

const columnHelper = createColumnHelper<LibraryListItem>();

const getColumns = ({
    handleSelect,
    handleRemove,
    selected,
    translations,
}: {
    handleSelect: (data: SelectedLibrary) => void;
    handleRemove: (id: string | number) => void;
    selected: SelectedLibrary;
    translations: { [id: string]: string };
}) => [
    columnHelper.display({
        id: "actions",
        meta: { isPinned: true },
        cell: ({ row }) => {
            const {
                id,
                datasetId,
                name,
                dataCustodianId,
                dataCustodian,
                darEnabled,
                darTemplatePublished,
                cohortEnabled,
            } = row.original;
            return (
                <div style={{ textAlign: "center" }}>
                    <StyledCheckbox
                        checked={selected[id]?.selected}
                        onChange={(_e, value) =>
                            handleSelect({
                                [id]: {
                                    selected: value,
                                    datasetId,
                                    name,
                                    teamId: dataCustodianId,
                                    teamName: dataCustodian,
                                    darEnabled,
                                    darTemplatePublished,
                                    cohortEnabled,
                                },
                            })
                        }
                        size="large"
                        sx={{ p: 0 }}
                        iconSx={{ mr: 0 }}
                    />
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),

    columnHelper.display({
        id: "name",
        cell: ({ row: { original } }) => {
            const { datasetId, name } = original;
            const linkHref = `/${RouteName.DATASET_ITEM}/${datasetId}`;

            return (
                <Link href={linkHref}>
                    <div style={{ textAlign: "center" }}>{name}</div>
                </Link>
            );
        },
        header: () => <span>{translations.name}</span>,
        size: 150,
    }),

    columnHelper.display({
        id: "darEnabled",
        cell: ({
            row: {
                original: { darEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {darEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => (
            <TooltipIcon
                buttonSx={{ p: 0 }}
                size="small"
                label={translations.darEnabled}
                content={translations.darEnabled}
            />
        ),
        size: 100,
    }),

    columnHelper.display({
        id: "cohortEnabled",
        cell: ({
            row: {
                original: { cohortEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {cohortEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => <span>{translations.cohortEnabled}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "dataCustodian",
        cell: ({
            row: {
                original: { dataCustodian },
            },
        }) => <div style={{ textAlign: "center" }}>{dataCustodian}</div>,
        header: () => <span>{translations.dataCustodian}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "entityType",
        cell: ({
            row: {
                original: { entityType },
            },
        }) => <div style={{ textAlign: "center" }}>{entityType}</div>,
        header: () => <span>{translations.entityType}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "delete",
        cell: ({ row: { original } }) => {
            return (
                <div style={{ textAlign: "center" }}>
                    <IconButton onClick={() => handleRemove(original.id)}>
                        <DeleteForeverIcon color="primary" />
                    </IconButton>
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),
];

export { getColumns };
