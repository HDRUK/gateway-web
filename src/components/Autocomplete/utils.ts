import { ValueType } from "./Autocomplete";

const getChipLabel = (
    options: { value: string | number; label: string }[],
    value: ValueType
) => {
    if (typeof value === "string") {
        return value;
    }

    return options.find(option => option.value === value)?.label;
};

export { getChipLabel };
