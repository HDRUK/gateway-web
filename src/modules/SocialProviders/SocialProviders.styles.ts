import { css } from "@emotion/react";
import { colors } from "@/config/theme";

export const dot = css({
    position: "absolute",
    width: "8px",
    height: "8px",
    right: "12px",
    top: "13px",
    borderRadius: "4px",
    backgroundColor: colors.orange,
});
