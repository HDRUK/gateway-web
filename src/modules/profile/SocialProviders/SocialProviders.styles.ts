import { css } from "@emotion/react";

export const dot = theme =>
    css({
        position: "absolute",
        width: "8px",
        height: "8px",
        right: "12px",
        top: "13px",
        borderRadius: "4px",
        backgroundColor: theme.palette.colors.orange,
    });
