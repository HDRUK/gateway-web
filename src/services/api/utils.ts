import { NotificationOptions } from "@/interfaces/Api";
import { AxiosResponse } from "axios";
import { Error } from "@/interfaces/Error";
import notificationService from "../notification";

interface ErrorNotificationProps {
    props: NotificationOptions;
    errorResponse: AxiosResponse<Error>;
    method: "delete" | "post" | "put" | "get";
}

const errorNotification = ({
    errorResponse,
    method,
    props,
}: ErrorNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const { data, status } = errorResponse || {};

    const fallbackTitle = i18n.exists(`api:common.error.status.${status}`)
        ? t(`api:common.error.status.${status}`)
        : "There has been an error";

    const title = data?.title || fallbackTitle;
    const message =
        data?.message ||
        t(`api:common.error.${method}.message`, {
            item: props.itemName || "Item",
        });

    notificationService.apiError(message, {
        title,
        message,
        errors: data?.errors,
        ...notificationProps,
    });
};

interface SuccessNotificationProps {
    props: NotificationOptions;
    method: "delete" | "post" | "put";
}

const successNotification = ({ props, method }: SuccessNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const customMessage = `api:${props.localeKey}.success.${method}.message`;
    const shouldOverideMessage = i18n.exists(customMessage);

    const message = shouldOverideMessage
        ? t(customMessage)
        : t(`api:common.success.${method}.message`, {
              item: props.itemName || "Item",
          });

    notificationService.success(message, { ...notificationProps });
};

export { errorNotification, successNotification };
