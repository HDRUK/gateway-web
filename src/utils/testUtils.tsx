import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
