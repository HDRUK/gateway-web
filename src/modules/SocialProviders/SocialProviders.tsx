/** @jsxImportSource @emotion/react */

import Box from "@/components/Box";
import Label from "@/components/Label";

import Image from "next/image";

import { colors } from "@/config/theme";
import * as styles from "./SocialProviders.styles";

interface SocialProvidersProps {
    label: string;
    name: string;
    getValues: (name: string) => unknown;
}

const SocialProviders = ({ label, getValues, name }: SocialProvidersProps) => {
    const socialProviders = [
        { label: "Google", value: "google", image: "google-logo.png" },
        { label: "LinkedIn", value: "linkedIn", image: "linkedIn-logo.png" },
        { label: "Microsoft", value: "microsoft", image: "microsoft-logo.png" },
    ];
    return (
        <Box sx={{ padding: 0, marginBottom: 2 }}>
            <Label name={name} label={label} sx={{ marginBottom: "4px" }} />
            <Box sx={{ padding: 0, display: "flex", gap: "10px" }}>
                {socialProviders.map(provider => (
                    <Box
                        key={provider.value}
                        sx={{
                            cursor: "pointer",
                            background: "white",
                            opacity:
                                getValues(name) === provider.value ? 1 : 0.5,
                            border: `2px solid ${colors.grey400}`,
                            borderRadius: "4px",
                            display: "flex",
                            gap: "12px",
                            padding:
                                getValues(name) === provider.value
                                    ? "6px 32px 6px 14px"
                                    : "6px 14px",
                            fontSize: "14px",
                            position: "relative",
                        }}>
                        <Image
                            src={`/images/logos/${provider.image}`}
                            alt={provider.label}
                            width="20"
                            height="20"
                        />
                        {getValues(name) === provider.value && (
                            <span css={styles.dot} />
                        )}
                        <div>Sign in with {provider.label}</div>
                    </Box>
                ))}
            </Box>
        </Box>
    );
};

export default SocialProviders;
