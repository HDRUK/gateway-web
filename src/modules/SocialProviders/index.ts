import SocialProviders from "./SocialProviders";

export default SocialProviders;
