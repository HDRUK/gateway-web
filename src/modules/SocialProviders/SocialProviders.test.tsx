import { useForm } from "react-hook-form";
import SocialProviders from "@/modules/SocialProviders";
import { render } from "@/utils/testUtils";

describe("SocialProviders", () => {
    const Component = () => {
        const { getValues } = useForm<{ provider: string }>({
            defaultValues: {
                provider: "google",
            },
        });
        return (
            <SocialProviders
                label="Your preferred sign in method"
                name="provider"
                getValues={getValues}
            />
        );
    };
    it("should render content", () => {
        const wrapper = render(<Component />);
        expect(wrapper.container).toMatchSnapshot();
    });
});
