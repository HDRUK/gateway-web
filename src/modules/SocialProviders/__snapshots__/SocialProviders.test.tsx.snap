// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialProviders should render content 1`] = `
<div>
  <div
    class="MuiBox-root mui-style-v1crop"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 mui-style-1dn6yst-MuiTypography-root-Label"
    >
      <label
        for="provider"
      >
        Your preferred sign in method
      </label>
    </p>
    <div
      class="MuiBox-root mui-style-sg1m7l"
    >
      <div
        class="MuiBox-root mui-style-z8xd21"
      >
        <img
          alt="Google"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Flogos%2Fgoogle-logo.png&w=48&q=75"
          srcset="/_next/image?url=%2Fimages%2Flogos%2Fgoogle-logo.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Flogos%2Fgoogle-logo.png&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
        <span
          class="mui-style-18qh7jl"
        />
        <div>
          Sign in with 
          Google
        </div>
      </div>
      <div
        class="MuiBox-root mui-style-1cgyyae"
      >
        <img
          alt="LinkedIn"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Flogos%2FlinkedIn-logo.png&w=48&q=75"
          srcset="/_next/image?url=%2Fimages%2Flogos%2FlinkedIn-logo.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Flogos%2FlinkedIn-logo.png&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
        <div>
          Sign in with 
          LinkedIn
        </div>
      </div>
      <div
        class="MuiBox-root mui-style-1cgyyae"
      >
        <img
          alt="Microsoft"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Flogos%2Fmicrosoft-logo.png&w=48&q=75"
          srcset="/_next/image?url=%2Fimages%2Flogos%2Fmicrosoft-logo.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Flogos%2Fmicrosoft-logo.png&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
        <div>
          Sign in with 
          Microsoft
        </div>
      </div>
    </div>
  </div>
</div>
`;
